lib/document.js

```diff
@@ -857,10 +859,10 @@ Document.prototype.updateOne = function updateOne(doc, options) {
   const query = this.constructor.updateOne({ _id: this._doc._id }, doc, options);
   const self = this;
   query.pre(async function queryPreUpdateOne() {
-    const res = await self._execDocumentPreHooks('updateOne', self);
+    const res = await self._execDocumentPreHooks('updateOne', options, self);
```

Comment:
This looks like a breaking change: `pre('updateOne', query => { ... })` will assume that the first parameter to pre('updateOne') is a query, not options.